pyFV3/stencils/dyn_core.py

```diff
@@ -51,7 +51,7 @@
 from pyFV3.stencils.riem_solver_c import NonhydrostaticVerticalSolverCGrid
 
 
-HUGE_R = 1.0e40
+HUGE_R = Float(1.0e40)
```

Comment:
This is incorrect. The value of HUGE_R is different - not a cast - depending on precision. Let me dig that for you

Reply:
From https://github.com/GEOS-ESM/GFDL_atmos_cubed_sphere/blob/700e6814005a3ad0341608430d9abe766b93219e/model/dyn_core.F90#L208
<html>
<body>
<!--StartFragment-->
#ifdef OVERLOAD_R4
--
208 | real,    parameter:: huge_r = 1.E8
209 | #else

<!--EndFragment-->
</body>
</html>

As a rule of thumb we need to always refer to original Fortran for things like that, and in case of 32-bit precision that means NASA's fork.

Reply:
by the way, does Python support auto type conversion like c++, huge_r sounds to me is not an issue for mixed precision capability in pace

Reply:
It does. It upcast by default.

HUGE_R will be an issue if we don't replicate the Fortran for validation of extreme cases if we ever get there, hence making sure we follow the original Fortran.